coderd/inboxnotifications.go

```diff
@@ -219,6 +219,12 @@ func (api *API) watchInboxNotifications(rw http.ResponseWriter, r *http.Request)
 	encoder := wsjson.NewEncoder[codersdk.GetInboxNotificationResponse](conn, websocket.MessageText)
 	defer encoder.Close(websocket.StatusNormalClosure)
 
+	// Log the request immediately instead of after it completes.
+	requestLogger := httpmw.RequestLoggerFromContext(ctx)
+	if requestLogger != nil {
```

Comment:
Rather than checking this everywhere you use it, you should just check for nil in the getter function you made. If tests are failing because it's not there, they should be updated to include this middleware in the stack